src/extension-runners/chromium.js

```diff
@@ -127,8 +127,58 @@ export class ChromiumExtensionRunner {
       chromeFlags.push(...this.params.args);
     }
 
-    if (this.params.chromiumProfile) {
-      chromeFlags.push(`--user-data-dir=${this.params.chromiumProfile}`);
+    function isUserDataDir(dirCandidate: string): boolean {
+      const localStateCandidate = path.join(dirCandidate, 'Local State');
+      const defaultCandidate = path.join(dirCandidate, 'Default');
+      // Local State and Default are typical for the user-data-dir
+      return fs.pathExistsSync(localStateCandidate)
+        && fs.pathExistsSync(defaultCandidate);
+
+    }
+
+    function isProfileDir(dirCandidate: string): boolean {
+      const securePreferencesCandidate = path.join(
+        dirCandidate,
+        'Secure Preferences');
+      //Secure Preferences is typical for a profile dir
+      return fs.pathExistsSync(securePreferencesCandidate);
+    }
+
+    let userDataDir = false;
+    const chromiumProfile = this.params.chromiumProfile;
+    if (chromiumProfile) {
+      const isProfileDirAndNotUserData =
+        fs.pathExistsSync(chromiumProfile)
+        && isProfileDir(chromiumProfile)
+        && !isUserDataDir(chromiumProfile);
+
+      if (isProfileDirAndNotUserData) {
+        const pathParts = path.parse(chromiumProfile);
+        const profileDirName = pathParts.base;
+        if (this.params.keepProfileChanges) {
+          // now check if the parent dir is a user-data-dir
+          if (!isUserDataDir(pathParts.dir)) {
+            throw new Error('The profile you provided is not in a ' +
+              'user-data-dir. The changes cannot be kept. Please either ' +
+              'remove --keep-profile-changes or use a profile in a ' +
+              'user-data-dir directory');
+          }
+          userDataDir = pathParts.dir;
+
+        } else {
+          // the user provided an existing profile directory but doesn't want
+          // the changes to be kept. we copy this directory to a temporary
+          // user data dir.
+          userDataDir = Launcher.prototype.makeTmpDir();
+          // copy profile dir to this temp user data dir.
+          fs.copySync(chromiumProfile, path.join(
+            userDataDir,
+            profileDirName));
+        }
+        chromeFlags.push(`--profile-directory='${profileDirName}'`);
+      } else {
+        userDataDir = chromiumProfile;
```

Comment:
Prepend comment to explain what `chromiumProfile` means at this point, so that the reader doesn't have to scroll all the way up.

And this doesn't properly handle the `--keep-profile-changes` case - see my top-level review comment for suggestions on improving this.

Reply:
I made an early return statement for it, so it should be clear now